datafusion-examples/examples/flight_sql_server.rs

```diff
@@ -0,0 +1,618 @@
+    ProstMessageExt, SqlInfo, TicketStatementQuery,
+};
+use arrow_flight::utils::batches_to_flight_data;
+use arrow_flight::{
+    Action, FlightData, FlightDescriptor, FlightEndpoint, FlightInfo, HandshakeRequest,
+    HandshakeResponse, IpcMessage, SchemaAsIpc, Ticket,
+};
+use arrow_schema::Schema;
+use dashmap::DashMap;
+use datafusion::logical_expr::LogicalPlan;
+use datafusion::prelude::{DataFrame, ParquetReadOptions, SessionConfig, SessionContext};
+use futures::{stream, Stream};
+use log::info;
+use mimalloc::MiMalloc;
+use prost::Message;
+use std::pin::Pin;
+use std::sync::Arc;
+use tonic::metadata::MetadataValue;
+use tonic::transport::Server;
+use tonic::{Request, Response, Status, Streaming};
+use uuid::Uuid;
+
+#[global_allocator]
+static GLOBAL: MiMalloc = MiMalloc;
+
+macro_rules! status {
+    ($desc:expr, $err:expr) => {
+        Status::internal(format!("{}: {} at {}:{}", $desc, $err, file!(), line!()))
+    };
+}
+
+/// This example shows how to wrap DataFusion with `FlightSqlService` to support connecting
+/// to a standalone DataFusion-based server with a JDBC client, using the open source "JDBC Driver
+/// for Arrow Flight SQL".
+///
+/// To install the JDBC driver in DBeaver for example, see these instructions:
+/// https://docs.dremio.com/software/client-applications/dbeaver/
+/// When configuring the driver, specify property "UseEncryption" = false
+///
+/// JDBC connection string: "jdbc:arrow-flight-sql://127.0.0.1:50051/"
+///
+/// Based heavily on Ballista's implementation: https://github.com/apache/arrow-ballista/blob/main/ballista/scheduler/src/flight_sql.rs
+/// and the example in arrow-rs: https://github.com/apache/arrow-rs/blob/master/arrow-flight/examples/flight_sql_server.rs
+///
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    env_logger::init();
+    let addr = "0.0.0.0:50051".parse()?;
+    let service = FlightSqlServiceImpl {
+        contexts: Default::default(),
+        statements: Default::default(),
+        results: Default::default(),
+    };
+    info!("Listening on {addr:?}");
+    let svc = FlightServiceServer::new(service);
+
+    Server::builder().add_service(svc).serve(addr).await?;
+
+    Ok(())
+}
+
+pub struct FlightSqlServiceImpl {
+    contexts: Arc<DashMap<String, Arc<SessionContext>>>,
+    statements: Arc<DashMap<String, LogicalPlan>>,
+    results: Arc<DashMap<String, Vec<RecordBatch>>>,
+}
+
+impl FlightSqlServiceImpl {
+    async fn create_ctx(&self) -> Result<String, Status> {
+        let uuid = Uuid::new_v4().hyphenated().to_string();
+        let session_config = SessionConfig::from_env()
+            .map_err(|e| Status::internal(format!("Error building plan: {e}")))?
+            .with_information_schema(true);
+        let ctx = Arc::new(SessionContext::with_config(session_config));
+
+        ctx.register_parquet(
+            "nadt_pq",
+            "/home/kmitchener/dev/convert/test-parquet/part-0.parquet",
```

Comment:
This seems like maybe it should be something less specific to your environment. Maybe it could be passed in as an environment variable?

Reply:
copy/paste error :) I put it back to referencing test data, thanks